compiler/luci/pass/src/FuseGRUPass.cpp

```diff
@@ -0,0 +1,674 @@
+ *  - 1 CircleReshape node;
+ *  - 1 CircleSub node;
+ *  - 1 CircleTanh node;
+ *  - 6 CircleSplitOut nodes;
+ *  - 5 CircleInput nodes;
+ *  - 5 CircleOutput nodes;
+ *
+ *  Body Graph:
+ *  [In_1]                     [In_2]--->[Add_2 (with Const)]--->[Out_2]            [In_3]
+ *    |    \                                                      |                   |
+ *    |     \                                         [In_4]---[Gather]      [Add_3 (with Const)]
+ *    |     [FullyConnected_1]                          |         |                   |
+ *    |               |                              [Out_4]      |                [Out_3]
+ *    |           [Split_1]                                 [FullyConnected_2]
+ *    |         /         |   \                                       |
+ *    |        |          |    \                                   [Split_2]
+ *    |    [Add_1] -------+----+---------------------------------/    |     |
+ *    |         |         |    |                                      |     |
+ *    |         |         |    ------------------------------------[Add_4]  |
+ *    |         |         |                                           |     |
+ *    |         |         |                                   [Logistic_1]  |
+ *    |         |         |                                          |      |
+ *    |         |         ----------------------------------------[Mul_2]   |
+ *    |         |                                                    \      /
+ *    |         |                                                     [Add_5]
+ *    |         |                                                        |
+ *    |      [Logistic_2]                                              [Tanh]
+ *     \     /           \                                               |
+ *      [Mul_1]          [Sub (with const)]                              |
+ *       \                                 \                             |
+ *        \                                 ---------------------------[Mul_3]
+ *         \                                                           /
+ *          \                                                         /
+ *           --------------------[Add_6]------------------------------
+ *                              /     \
+ *                             /       \
+ *                         [Reshape]   [Out_5]
+ *                             |
+ *                           [Out_1]
+ */
+class GRUPattern1 final : public GRUPatternBase
+{
+public:
+  GRUPattern1(luci::CircleWhileOut *candidate) : GRUPatternBase(candidate)
+  {
+    assert(candidate);
+    _while_out_node = candidate;
+  }
+
+public:
+  bool matched() override;
+};
+
+bool GRUPattern1::matched()
+{
+  // 0 - check while node
+  _while_node = dynamic_cast<luci::CircleWhile *>(_while_out_node->input());
+  if (_while_node == nullptr)
+    return false;
+
+  // 1 - check condition graph: only one Less operation
```

Comment:
`only one Less operation` and below condition doesn't match.
- 1/ fix comment like `Less operation should exist`
- 2/ fix implementation to check only one `Less` exist